File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day04Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day04Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day04Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day04();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void getSumOfAllCardPoints() {
        var instance = new Day04();
        int expected = 13;
        var actual = instance.getSumOfAllCardPoints(lines);
        assertEquals(expected, actual, "The sum of all ScratchCard points " +
                "is not what was expected: " + expected);
    }

    @Test
    void getTotalNumberOfCards() {
        var instance = new Day04();
        int expected = 30;
        var actual = instance.getTotalNumberOfCards(lines);
        assertEquals(expected, actual, "The total number of ScratchCards " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day04Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day04Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day04Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day04()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testCalculateOccurrencesOfXmas() {
    val instance = Day04()
    val expected = 18

    val actual = instance.calculateOccurrencesOfXmas(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual occurrences of $actual is not the expected " +
            "number of occurrences of $expected."
    )
  }

  @Test
  fun testCalculateOccurrencesOfMas() {
    val instance = Day04()
    val expected = 9

    val actual = instance.calculateOccurrencesOfMas(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual occurrences of $actual is not the expected " +
            "number of occurrences of $expected."
    )
  }
}
